src/shell/shell.cpp

```diff
@@ -1442,6 +1447,18 @@ void SHELL_Init() {
 	                             "DOS version %d.%02d\n");
 	MSG_Add("SHELL_CMD_VER_INVALID", "The specified DOS version is not correct.\n");
 
+	// Setup Help
+	for (const auto &c : DOS_Shell::shell_cmds) {
+		auto filter = c.second.flags == 1 ? HelpUtil::Filter::COMMON
+		                                  : HelpUtil::Filter::ALL;
+		HelpUtil::add_to_help_list(c.first,
+		                           HelpUtil::Detail{filter,
+		                                            c.second.category,
+		                                            HelpUtil::CmdType::SHELL,
+		                                            c.second.help},
+		                           true);
```

Comment:
This is looking very name-space heavy. Can we drop the `HelpUtil::` namespace and if needed, put a bit more of that Help-relative naming into the enums (if we really think it's needed?).

Reply:
I could do `using namespace HelpUtil;` here as well, but based on your other comment, it seems I'll be dropping the namespace entirely.

Reply:
More about sticking to the current style, `MODULE_FuncName` where the prefix acts to differentiate the API calls 

With a full refactoring, I could see something like
```c++
 Namespace MODULE {
    FuncName
}
```

Doing the same job. 

But the number of shared DOSBox functions across sources is pretty small, so this feels a bit overkill (given the current design is a decent surrogate). 

If we had ambiguous function names, then yeah, it would be in more dire shape for namespaces (I think, anyway)

Reply:
@shermp, 
Thinking more about the shared enums that cross into multiple sources.. their naming conventions isn't as consistent and tidy as the shared functions (MODULE_SomeFunc). 

**`git grep enum include/*.h | grep .h | grep {`**

```text
bios_disk.h:    enum { NONE,READ,WRITE } 
dma.h:enum class DMA_DIRECTION { READ, WRITE };
dos_inc.h:enum { STDIN=0,STDOUT=1,STDERR=2,STDAUX=3,STDPRN=4};
dos_inc.h:enum COUNTRY {
dos_system.h:   enum { NONE,READ,WRITE } 
dos_system.h:   enum TDirSort { NOSORT, ALPHABETICAL, DIRALPHABETICAL, ALPHABETICALREV, DIRALPHABETICALREV };
dos_system.h:enum { OPEN_READ=0, OPEN_WRITE=1, OPEN_READWRITE=2, OPEN_READ_NO_MOD=4, DOS_NOT_INHERIT=128};
dos_system.h:enum { DOS_SEEK_SET=0,DOS_SEEK_CUR=1,DOS_SEEK_END=2};
dosbox.h:enum SVGACards {
dosbox.h:enum MachineType {
fpu.h:enum FPU_Tag {
fpu.h:enum FPU_Round {
hardware.h:enum OPL_Mode {
joystick.h:enum JoystickType {
keyboard.h:enum KBD_KEYS {
logging.h:enum LOG_TYPES {
logging.h:enum LOG_SEVERITIES {
mixer.h:enum BlahModes {
mixer.h:enum MixerModes {
mixer.h:enum LINE_INDEX : uint8_t {
mixer.h:enum class FilterState { Off, On, ForcedOn };
sdlmain.h:enum MouseControlType {
sdlmain.h:enum SCREEN_TYPES     {
sdlmain.h:enum class FRAME_MODE {
sdlmain.h:enum class HOST_RATE_MODE {
sdlmain.h:enum class SCALING_MODE { NONE, NEAREST, PERFECT };
sdlmain.h:enum class VSYNC_STATE {
sdlmain.h:enum PRIORITY_LEVELS {
serialport.h:enum SERIAL_PORT_TYPE { // 
setup.h:                enum Value { Always, WhenIdle, OnlyAtStart, Deprecated };
shell.h:        enum class HELP_LIST { ALL, COMMON };
support.h:enum class ResourceImportance { Mandatory, Optional };
vga.h:enum VGAModes {
vga.h:enum Drawmode { PART, DRAWLINE, EGALINE };
vga.h:enum class VGA_RATE_MODE { DEFAULT, HOST, CUSTOM };
vga.h:enum EGAMonitorMode { CGA, EGA, MONO };
```

A fair number are universal: log severity, VGA mode, machine type, file IO mode, etc. But other are pretty specific too. 

But the the other thing that jumped out at me is that there isn't much of a naming convention (granted.. this is 20+ years of organic growth). 

So your use of namespaces was one approach (I still thing namespacing enums, especially class enums, looks pretty verbose). But can you think of a convention that we can apply more strictly? (And ideally, start to refactor the older enums into this new style?) 

Would be nice to get the enums more standardized like the function API is.

Reply:
`using namespace ...` does help cut down on the long names somewhat, but I don't really have any further ideas at this point.

Reply:
Namespace gone.